### PR TITLE
Trim commit message to not include description

### PR DESCRIPTION
**Pull request by Methuselah96**
Resolves https://github.com/Azure/webapps-deploy/issues/192.

I considered trimming the commit message to a certain length, but it appears that it would be hard to completely predict how long the message sent to Kudu is since the message contains more than just the commit message. It seems reasonable to just include the first line of the commit message.

### CODE REVIEW

**Comment by mattnieland** [comment]
ANYONE gonna do anything about this problem and approve his PR?

**Comment by DavidBoike** [comment]
Can someone please look at this PR? This problem is a constant annoyance.

**Comment by DavidBoike** [comment]
@Methuselah96 is there any way I can help? I can't even see why the CI failed because it was run so long ago, and I can't trigger it to re-run. Or are you only waiting for the Azure team to look at the PR?

**Comment by DavidBoike** [comment]
I tried opening https://github.com/Azure/webapps-deploy/pull/243 with the exact same changes (`npm run build` was run and `npm run test` passes all tests) in order to see what might be in the problem with the CI, but that results in "Waiting for review: automation test needs approval to start deploying changes." so there doesn't seem to be any point in having that PR.

It seems all this PR really needs is attention.

**Comment by Methuselah96** [comment]
@DavidBoike Yeah, I'm just waiting on review, not sure what's up with the CI.

**Review by darena-patrick — APPROVED** [review]
lgtm 🎉

**Comment by shubham805** [comment]
@Methuselah96 How did you reproduce the 404 error (https://github.com/Azure/webapps-deploy/issues/192) with a long commit message?

**Comment by Methuselah96** [comment]
@shubham805 The situation where we run into this is as described [here](https://github.com/Azure/webapps-deploy/issues/192#issuecomment-999662040). If you want to reproduce it without squash merge, you can just include a really long multi-line commit message (that includes a one line summary and then a really long description).

**Review by shpraka — APPROVED** [review]

**Review by shubham805 — APPROVED** [review]

**Comment by shubham805** [comment]
@Methuselah96 The PR looks good. Can you add a dummy commit just to re trigger the pr check pipelines?

**Comment by Methuselah96** [comment]
@shubham805 Done, says the workflow is awaiting approval.

**Comment by shubham805** [comment]
@Methuselah96 There were some issues in the PR check pipeline. We have updated it. Can you again add a new dummy commit, it will pick up the updated workflow file.

**Comment by Methuselah96** [comment]
@shubham805 Done, awaiting approval again.

**Comment by Methuselah96** [comment]
@shubham805 Are there plans to release this?

**Comment by shpraka** [comment]
@Methuselah96 this is released. Do let us know in case you find any issues.

**Comment by IGx89** [comment]
We just ran into this on v2 -- appears to maybe not be fixed for every scenario? Using a self-hosted runner. The ZIP deploy URL was 2572 characters long...

**Comment by IGx89** [comment]
Maybe the change should have been `.split(/(\\r)?\\n/)[0]` instead of `.split(/\r?\n/)[0]`? The latter doesn't work when I test it against a commit message such as "abc\nxyz\r\n123\n"

**Comment by Methuselah96** [comment]
This was replaced by https://github.com/Azure/webapps-deploy/pull/271 anyway, so the code in this PR is no longer used (although I agree it's wrong).

**Comment by IGx89** [comment]
Should I submit a new issue, about 2572 characters being too long? #271 assumes that up to 8000 characters is acceptable, which from our current experience seems inaccurate.

**Comment by Methuselah96** [comment]
Yeah, that would be great, thanks!

**Comment by IGx89** [comment]
Done: #288

**Pull request merged**
